.github/workflows/code_quality.yaml

```diff
@@ -84,15 +82,33 @@ jobs:
   ruff:
     name: ruff
     runs-on: ubuntu-latest
+    env:
+      poetry_version: '1.7.1'
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: Cache poetry in ~/.local
+        uses: actions/cache@v3
+        id: cached-home-local
+        with:
+          path: ~/.local
+          key: ${{ runner.os }}-local-${{ env.poetry_version }}
```

Comment:
what is the benefit of this step?

Reply:
It allows skipping pip install poetry (the next step) if restoring cache succeeds. Restoring poetry from cache this way takes a few seconds less than installing with pip every time.

So, it saves 3 to 5 seconds off of the job runtime.